### DISCUSSION TITLE
sending multiple RCP codes

### DISCUSSION THREAD

**Discussion by AprilArcus** [kovidgoyal/kitty#3992] [Q&A] [answered]
My goal is to reset the configured colors before setting the configured colors, in a single RCP command. The command string I'm generating looks like:

`\x1bP@kitty-cmd{"cmd":"set-colors","version":[0,23,1],"no_response":"true","payload":{"reset":true}}\x1b\\x1bP@kitty-cmd{"cmd":"set-colors","version":[0,23,1],"no_response":"true","payload":{"colors":{"color0":4210241,"color1":16736648,"color2":11132022,"color3":16767078,"color4":16554087,"color5":11247090,"color6":7920872,"color7":16579834,"color8":7499890,"color9":16736648,"color10":11132022,"color11":16767078,"color12":16554087,"color13":11247090,"color14":7920872,"color15":16579834,"foreground":16579834,"background":4210241,"cursor":16579834,"selection_background":7829110},"cursor_text_color":16579834}}\x1b\`

However, only the `reset` command is respected; the color configuration command has no effect. If I generate two commands with two separate printfs and then eval that string, everything works. Is it necessary to wait some period of time between successive commands, e.g. a tick of the REPL cycle?

**Comment by kovidgoyal** [comment, marked as answer]
IIRC if you specify reset everything else is ignored. So you have to use
two successive commands.

**Reply by AprilArcus** [reply]
Okay, thanks. This is surprising because the xterm-style OSC codes (which kitty also respects) behave differently. This sequence:

`\x1b]104\x07\x1b]110\x07\x1b]112\x07\x1b]117\x07\x1b]111\x07\x1b]4;0;#3b3c35;1;#f92672;2;#a6e22e;3;#e6db74;4;#fd971f;5;#ae81ff;6;#66d9ef;7;#fdfff1;8;#6e7066;9;#f92672;10;#a6e22e;11;#e6db74;12;#fd971f;13;#ae81ff;14;#66d9ef;15;#fdfff1\x07\x1b]10;#fdfff1\x07\x1b]12;#fdfff1\x07\x1b]17;#75766e\x07\x1b]11;#3b3c35\x07`

will reset the theme colors and then set them.
